src/mentionPlugin/MentionSearch/index.js

```diff
@@ -14,12 +14,16 @@ export default (mentions) => {
     };
 
     // Get the first 5 mentions that match
-    getMentionsForFilter = () => (
-      mentions
+    getMentionsForFilter = () => {
 
-      // TODO better search algorithm than startsWith
-      // mentions.filter((mention) => mention.get('handle').startsWith(this.mentionSearch)).slice(0, 10)
-    );
+      const userEntered = this.props.editor.props.editorState.getCurrentContent().getLastBlock().getText();
```

Comment:
Filtering works well!

It breaks when the `@` mention is not happening at the end. See: 
![filter](https://cloud.githubusercontent.com/assets/223045/13555434/349d6d80-e375-11e5-8259-c0573c852a41.gif)